src/reyna/Connection/ConnectionInfo.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace Reyna
+{
+    using System;
+    using System.IO;
+    using OpenNETCF.Net.NetworkInformation;
+
+    public class ConnectionInfo : IConnectionInfo
+    {
+        public bool Connected
+        {
+            get
+            {
+                try
+                {
+                    foreach (var ni in NetworkInterface.GetAllNetworkInterfaces())
+                    {
+                        if (string.Compare(ni.CurrentIpAddress.ToString().Trim(), "0.0.0.0", true) != 0)
+                        {
+                            return true;
+                        }
+                    }
+                }
+                catch (Exception)
+                { 
+                }
+                
+                return false;
+            }
+        }
+
+        public bool Mobile
+        {
+            get
+            {
+                try
+                {
+                    var interfaces = NetworkInterface.GetAllNetworkInterfaces();
+                    bool mobileNetworkConnected = false;
+                    bool otherNetworksConnected = false;
+                    foreach (var ni in interfaces)
+                    {
+                        if (GPRSNetwork(ni))
```

Comment:
Can that even be true, if CurrentIpAddress than is 0.0.0.0? Also would be slightly more intuitive for me to first check ip-address if connected at all and then do the if-else on mobile or other.